lib/provisioning.js

```diff
@@ -245,16 +249,16 @@ function detect(options, callback) {
 			}
 		}
 
-		fs.exists(profileDir, function (exists) {
-			exists && fs.readdirSync(profileDir).forEach(function (name) {
+		for (const profileDir of profileDirs) {
+			fs.readdirSync(profileDir).forEach(function (name) {
 				ppRegExp.test(name) && parseProfile(path.join(profileDir, name));
 			});
+		}
```

Comment:
We need the exists check, but we can just use `existsSync()`:

```suggestion
		for (const profileDir of profileDirs) {
			if (fs.existsSync(profileDir)) {
				for (const name of fs.readdirSync(profileDir)) {
					if (ppRegExp.test(name)) {
						parseProfile(path.join(profileDir, name));
					}
				}
			}
		}
```

Reply:
That is already done in `getExistingProvisioningProfileDirectories`, specifically in [#L397](https://github.com/tidev/ioslib/blob/fix/new-provisioning-profile-directory/lib/provisioning.js#L397)

Reply:
Ah, sorry!